infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/plugins/CorfuReplicationClusterManagerAdapter.java

```diff
@@ -1,51 +1,59 @@
 package org.corfudb.infrastructure.logreplication.infrastructure.plugins;
 
-import lombok.Getter;
-
 import org.corfudb.infrastructure.logreplication.infrastructure.CorfuReplicationDiscoveryServiceAdapter;
 import org.corfudb.infrastructure.logreplication.proto.LogReplicationClusterInfo.TopologyConfigurationMsg;
 
-public abstract class CorfuReplicationClusterManagerAdapter {
-    @Getter
-    CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService;
+/**
+ * This is the interface for CorfuReplicationClusterManager.
+ * Implementation of this should have following members:
+ * 1. corfuReplicationDiscoveryService that is needed to notify the cluster configuration change.
+ * 2. localEndpoint that has the local node information.
+ *
+ */
+public interface CorfuReplicationClusterManagerAdapter {
 
-    @Getter
-    TopologyConfigurationMsg topologyConfig;
+    /**
+     *   Register the discovery service and start the cluster manager callback service.
+     *   Implementation of this function should call register(), register this ClusterManager at the upper
+     *   layer to get the callback for topology change.
+     */
+     void start(CorfuReplicationDiscoveryServiceAdapter corfuReplicationDiscoveryService);
```

Comment:
Assuming that the start should call the register is a bit loose, as implementers might not do the right thing. Maybe it's better if we call from the DiscoveryService start (which could be more of a setDiscoveryService) and register.... and avoid issues coming from an implementer not really doing the register explicitly.

Reply:
+1.  My suggestion is to have a registerDiscoveryService() method here which is called by the discovery service to register itself.

Reply:
@xiaoqin2012 I was referring to this..

Reply:
This is out of date.